Assets/Scripts/Lobby/LobbyManager.cs

```diff
@@ -0,0 +1,76 @@
+using System.Collections;
+using System.Linq;
+using System.Text;
+using Characters;
+using UI;
+using Unity.Services.Lobbies;
+using UnityEngine;
+
+namespace Lobby
+{
+    public class LobbyManager : MonoBehaviour
+    {
+        [SerializeField] private PlayButton playButton;
+        private ICharacterLoader _characterLoader;
+        private Unity.Services.Lobbies.Models.Lobby CurrentLobby { get; set; }
+        private string lobbyId;
+        
+        public Unity.Services.Lobbies.Models.Lobby CreateLobby()
+        {
+            var task = Lobbies.Instance.CreateLobbyAsync("MyLobby", 4, new CreateLobbyOptions()
+            {
+                IsPrivate = true
+            });
+            var lobby = task.GetAwaiter().GetResult();
+            CurrentLobby = lobby;
+            lobbyId = lobby.Id;
+            StartCoroutine(GetLobby(2.5f));
+            return lobby;
+        }
+
+        public Unity.Services.Lobbies.Models.Lobby JoinLobby(string lobbyCode)
+        {
+            var task = Lobbies.Instance.JoinLobbyByCodeAsync(lobbyCode);
+            var lobby = task.GetAwaiter().GetResult();
+            CurrentLobby = lobby;
+            lobbyId = lobby.Id;
+            StartCoroutine(GetLobby(0.1f));
+            return lobby;
+        }
+        
+        private void GetLobby()
+        {
+            var task = Lobbies.Instance.GetLobbyAsync(lobbyId);
+            var lobby = task.GetAwaiter().GetResult();
```

Comment:
This line of code is going to wait for the result of an async method, effectively blocking the thread. Consider using `await` and make this method `async`.
```suggestion
        private async void GetLobby()
            var lobby = await Lobbies.Instance.GetLobbyAsync(lobbyId);
```